### ISSUE TITLE
Create a means by which users can add comments to a form being filled out.

### ISSUE THREAD

**Issue by SimonBiggs** [SimonBiggs/scriptedforms#73]
Comments should come in three colour options, info (blue), warning (orange), and critical (red). The cell box and then the resulting highlight around the Markdown results will have one of those three colours. There is a fourth option that produces a grey cell which is to pick a code cell.

The UI could be to have a clickable bar on the right side of the form. Moving the mouse over the bar causes a transparent cell to appear in the location where it would appear should you click on the bar at that location. The transparent cell will be inline/block so that you can see how the other items will shift around the cell you will be adding.

The cell that is inserted is a notebook cell. It defaults to being a markdown cell for the purpose of taking notes. But it would be quite neat if it could be switched to a code cell.

**Comment by SimonBiggs** [comment]
The user interface of the cell could maybe have four tabs up the right hand side. Each tab coloured according to its item with info as the default.

If the code tab on the cell is selected then a play button also appears at the bottom right hand corner of the cell.

**Comment by SimonBiggs** [comment]
Don't do this just yet. Or maybe don't do it at all. Focus on other things first.

**Comment by robmarkcole** [comment]
Something like comments in Word? I think Googles Collaboratory adds comments, not sure how its implemented

**Comment by SimonBiggs** [comment]
Well I was considering adding the ability to include full blown notebook cells by clicking in the margin... It would allow commenting of sorts without changing the underlying template.

I do think though this idea might be deviating too far from the goals of what I want to do. I'm just going to let this sit for a while and see what thoughts turn up.

**Comment by robmarkcole** [comment]
Agreed - let people use it in anger and see what comments come back
